### PR TITLE
Furmanets-Championship

### PR DESCRIPTION
**Pull request by Furmanets**

### CODE REVIEW

File: 2018/Furmanets/Championship/Championship/GrafGrid.cs [hunk 1]
```diff
@@ -0,0 +1,156 @@
+﻿using System.Collections.Generic;
+using ConsoleChampionship;
+
+namespace Championship
+{
+    public class GrafGrid
+    {
+        public List<string> Graf = new List<string>();
```
Comment (vikian050194) [inline 2018/Furmanets/Championship/Championship/GrafGrid.cs]: Change spelling of the word

Comment (vikian050194) [reply 2018/Furmanets/Championship/Championship/GrafGrid.cs]: Other names of variables and classes and methods

File: 2018/Furmanets/Championship/Championship/GrafGrid.cs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        {
+            _maxLengthName = maxLengthName;
+        }
+
+        public void ConstructionGraf(Tournament tournament)
+        {
+            var grid = tournament.Grid;
+            var distanceBetweenStage = 7;
+
+            for (var i = 0; i < grid.Tournament[0].Meetings.Count * 6; i++)
```
Comment (vikian050194) [inline 2018/Furmanets/Championship/Championship/GrafGrid.cs]: magic number

File: 2018/Furmanets/Championship/Championship/GrafGrid.cs [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+            {
+                Graf.Add(new string(' ', 2));
+            }
+
+            var previousLines = new List<int>();
+            var isFirsRound = true;
+
+            foreach (var round in grid.Tournament)
+            {
+                Graf[0] += new string(' ', _maxLengthName - 4) + GetForPrintRound(round.Stage) + new string(' ', distanceBetweenStage);
```
Comment (vikian050194) [inline 2018/Furmanets/Championship/Championship/GrafGrid.cs]: Too long string

File: 2018/Furmanets/Championship/Championship/GrafGrid.cs [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+                    continue;
+                }
+
+                IdentificationMeeting(meeting, out var nameLengthFirstPlayer, out var nameLengthSecondPlayer,
+                    out var goalsFisrtPlayer, out var goalsSecondPlayer);
+
+                Graf[meetingNumber] +=
+                    new string(' ', _maxLengthName - nameLengthFirstPlayer) + meeting.FirstPlayer + goalsFisrtPlayer;
+
+                Graf[meetingNumber + 1] += new string(' ', Graf[meetingNumber].Length - Graf[meetingNumber + 1].Length) + "|----";
```
Comment (vikian050194) [inline 2018/Furmanets/Championship/Championship/GrafGrid.cs]: Too long string

File: 2018/Furmanets/Championship/Championship/Tournamet.cs [hunk 5]
```diff
@@ -1,122 +1,100 @@
...
+                return;
             }
 
-            foreach (var meeting in Standings.Tournament)
+            for (var i = Grid.Tournament[1].Meetings.Count - 1; i >= countFirstStageMeetings / 2; i--)
             {
-                meeting.SecondPlayer = _players[indexPlayer];
-                indexPlayer++;
-                if (_players.Count == indexPlayer)
+                try
```
Comment (vikian050194) [inline 2018/Furmanets/Championship/Championship/Tournamet.cs]: Try not to use try-catch here

File: 2018/Furmanets/Championship/Championship/Tournamet.cs [hunk 6]
```diff
@@ -1,122 +1,100 @@
...
-                {
-                    PromoteWinner(meeting, meeting.FirstPlayer, nextRoundGrid);
+                    // meeting.SecondPlayer = "@" + meeting.SecondPlayer + "$";
+                    PromoteWinner(meeting, meeting.SecondPlayer);
                 }
             }
         }
 
-        private void PromoteWinner(Meeting meeting, string player, TournamentGrid nextRoundGrid)
+        private void PromoteWinner(Meeting meeting, string player)
```
Comment (vikian050194) [inline 2018/Furmanets/Championship/Championship/Tournamet.cs]: Strange name of method

File: 2018/Furmanets/Championship/Championship/Tournamet.cs [hunk 7]
```diff
@@ -3,120 +3,53 @@
...
-                if (meeting.Stage == "final")
-                {
-                    Draftsman.CongratulationWinner(meeting.SecondPlayer);
-                }
-                else
-                {
-                    PromoteWinner(meeting, meeting.FirstPlayer, nextRoundGrid);
-                }
+                PromoteWinner(meeting, meeting.SecondPlayer);
+                meeting.Winner = MeetingWinningIndicator.SecondPlayer;
```
Comment (chuvaginaviktoriya) [inline 2018/Furmanets/Championship/Championship/Tournamet.cs]: If we have equal scores the method automaticly considers second player to be the winner of meeting

File: 2018/Furmanets/Championship/Championship/Tournamet.cs [hunk 8]
```diff
@@ -3,120 +3,53 @@
 
 namespace Championship
 {
+    [Serializable]
     public class Tournament
     {
-        public TournamentGrid Standings;
-        private List<string> _players;
-        private int _playersCount;
+        public List<Round> TournamentRounds;
```
Comment (chuvaginaviktoriya) [inline 2018/Furmanets/Championship/Championship/Tournamet.cs]: These fields must not be public

File: 2018/Furmanets/Championship/ConsoleChampionship/GraphPainter.cs [hunk 9]
```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using Championship;
+
+namespace ConsoleChampionship
+{
+    abstract class GraphPainter
+    {
+        public abstract void PaintGraph(Tournament tournament);
+        
+        public virtual void WriteNamePlayer(Meeting meeting, bool isFirst)
```
Comment (chuvaginaviktoriya) [inline 2018/Furmanets/Championship/ConsoleChampionship/GraphPainter.cs]: Wrong order of words in method's name

File: 2018/Furmanets/Championship/ConsoleChampionship/GraphPainter.cs [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+
+                if (meeting.SecondPlayer != null && meeting.SecondPlayer.Length > maxLengthName)
+                {
+                    maxLengthName = meeting.SecondPlayer.Length;
+                }
+            }
+            return maxLengthName;
+        }
+
+        public string GetForPrintRound(int round)
```
Comment (chuvaginaviktoriya) [inline 2018/Furmanets/Championship/ConsoleChampionship/GraphPainter.cs]: Strange and confusing name "GetForPrintRound"

File: 2018/Furmanets/Championship/ConsoleChampionship/HorisontalGraphPainter.cs [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+
+namespace ConsoleChampionship
+{
+    internal class HorisontalGraphPainter : GraphPainter
+    {
+        public override void PaintGraph(Tournament tournament)
+        {
+            var nextCursorPositionTop = 3;
+            var nextCursorPositionLeft = 0;
+            var nextDistanceBeewinPalyers = 2;
```
Comment (chuvaginaviktoriya) [inline 2018/Furmanets/Championship/ConsoleChampionship/HorisontalGraphPainter.cs]: Wrong spelling of "between"

File: 2018/Furmanets/Championship/ConsoleChampionship/HorisontalGraphPainter.cs [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+                    if (meeting.FirstPlayer == null && meeting.SecondPlayer == null && round.Equals(tournament.TournamentRounds[0]))
+                    {
+                        continue;
+                    }
+
+                    Console.SetCursorPosition(positionCursorLeft, positionCursorTop);
+
+                    WriteNamePlayer(meeting, true);
+
+                    var wereDrowLine = distanceBetweenPlayers / 2;
```
Comment (chuvaginaviktoriya) [inline 2018/Furmanets/Championship/ConsoleChampionship/HorisontalGraphPainter.cs]: Strange and confusing name "wereDrowLine"

**Pull request merged**